include/zephyr/arch/xtensa/cache.h

```diff
@@ -177,7 +177,7 @@ static inline void __sparse_cache *arch_xtensa_cached_ptr(void *ptr)
  */
 static inline void *arch_xtensa_uncached_ptr(void __sparse_cache *ptr)
 {
-	return (void *)z_xtrpoflip((uint32_t) ptr,
+	return (void *)z_xtrpoflip((uint32_t)(__sparse_force void *)ptr,
```

Comment:
unrelated to this PR, but I think that https://github.com/zephyrproject-rtos/zephyr/blob/b1615880320d39b89e1b7b467d6b9e6b67d322be/include/zephyr/arch/xtensa/cache.h#L123 is wrong. It works with our current Xtensa versions, but it isn't generic. The condition checks that just one bit gets flipped in the pointer cache-uncache aliasing bits. This is a sufficient condition to decide that a conversion can be performed in one operation but not a necessary one. The minimal condition would be something like
```
if (!(rxor & rto) || !(rxor & rfrom))
```
and it seems to me that would also be a compile-time check?

Reply:
as for this specific change - it seems strange to me. Is it really fixing sparse warnings? Sparse complains about casting `void __sparse_cache *` to an integer type but doesn't complain any more after this modification?

Reply:
After changing line 729-730 in `mm_drv_intel_adsp_mtl_tlb.c`:
```
				POINTER_TO_UINT(z_soc_uncached_ptr(
					(__sparse_cache void *)UINT_TO_POINTER(phys_addr)));
```
to fix `warning: incorrect type in argument 1 (different address spaces)` due to pointer in different address spaces, sparse complains that
```
zephyr/include/zephyr/arch/xtensa/cache.h:180:37: warning: cast removes address space '__cache' of expression
```
Hence the need to force an address space change in `cache.h` here at line 180. It seems like there is an implicit cast to `(void *)` before converting to `uint32_t`.

Reply:
ok, I see, how about a single `(__sparse_force uint32_t)ptr` - would that work and fix the warning

Reply:
That works, and I have updated the PR with this.